lib/reviews.js

```diff
@@ -132,14 +138,31 @@ class ReviewAnalyzer {
     const collaborators = this.collaborators;
     for (const [ login, review ] of reviewers) {
       const reviewer = collaborators.get(login.toLowerCase());
-      if (review.state === APPROVED) {
+      if (review.state === APPROVED || this.isApprovedInComment(review)) {
         result.approved.push({ reviewer, review });
       } else if (review.state === CHANGES_REQUESTED) {
         result.rejected.push({ reviewer, review });
       }
     }
     return result;
   }
+
+  /**
+   * @param review
+   * @returns {boolean}
+   */
+  isApprovedInComment(review) {
+    return review.state === COMMENTED && this.hasLGTM(review, 'ref');
+  }
+
+  /**
+   * @param object
+   * @param prop: string
+   * @returns {boolean}
+   */
+  hasLGTM(object, prop) {
+    return LGTM_RE.test(object[prop]);
```

Comment:
So this is too loose. It would pass for `I'm revoking my LGTM`.
IMHO this should be `object[prop].toLowerCase() === 'lgtm'`.

Reply:
@refack This is already the case for normal comments, I think we should log an info here, like what we do for those, to remind the landers to double-check. The actual context would be too hard to parse without some kind of natural language processing..

Reply:
Good idea!

Reply:
OK not log an info here, but make a new type of `REVIEW_SOURCES`, maybe `FROM_REVIEW_COMMENT`, and warn them in `checkReviews` like what we do for `FROM_COMMENT`

Reply:
@refack re: `I'm revoking my LGTM.` they would've put an red cross there I think?

Reply:
I'm thinking of something I saw last week - https://github.com/nodejs/node/pull/16612#pullrequestreview-73478662

Reply:
That case was not a "Request Changes" , just "Don't count me as approver", same as dismissed review.

Reply:
Another idea: if we are uncertain about a LGTM, we should log it as `?Reviewed-By: xxx`, and then core-validate-commit should complain about it.

Reply:
On a second thought, I am finding the `object[prop].toLowerCase() === 'lgtm'` approach better (although we can just do that by changing `LGTM_RE` to `/^lgtm$/i` and trim the comments before testing them), it's just simpler that way. I think people who don't give LGTM either this way or with a green review would not mind not being listed in the review-by list.